### PR TITLE
Add support for Union types.

### PR DESCRIPTION
**Pull request by anntzer**
... as documented in features.rst.

### CODE REVIEW

**Comment by evanunderscore** [comment]
Thanks for this! Code looks good to me. You're welcome to merge when ready.

File: docs/features.rst [hunk 1]
```diff
@@ -172,6 +172,19 @@ has fields, convert each argument to the correct type, and wrap them into the
 annotation class.  When a `typing.NamedTuple` is used for an optional argument,
 the names of the fields are used in the help.
 
+Unions
+------
+
+Union types can be specified with ``typing.Union[type1, type2]``, or, when
+using docstring annotations, as ``type1 or type2``.  When an argument is
+annotated with an union type, an attempt is made to convert the command-line
```
Comment (evanunderscore) [inline docs/features.rst]: a* union

Comment (anntzer) [reply docs/features.rst]: edited

**Comment by anntzer** [comment]
Thanks for the review; self-merging.

**Pull request merged**